patches/minecraft/net/minecraft/enchantment/EnchantmentHelper.java.patch

```diff
@@ -27,3 +42,49 @@
              {
                  for (int i = enchantment.func_77325_b(); i > enchantment.func_77319_d() - 1; --i)
                  {
+@@ -429,6 +432,26 @@
+         return list;
+     }
+ 
++    /* ======================================== FORGE START =====================================*/
++
++    /**
++     * Calculate the damage the enchantments of the given ItemStack add to the attack on the given entity.
++     *
++     * @param stack the itemstack used for attacking
++     * @param entity the entity being attacked
++     * @return the additional damage to do
++     */
++    public static float getModifierForCreature(ItemStack stack, @javax.annotation.Nullable Entity entity)
++    {
++        field_77521_c.field_77495_a = 0.0F;
++        field_77521_c.entity = entity;
++        func_77518_a(field_77521_c, stack);
++        field_77521_c.entity = null;
++        return field_77521_c.field_77495_a;
++    }
++
++    /* ======================================== FORGE END =====================================*/
++
+     static final class DamageIterator implements EnchantmentHelper.IModifier
+         {
+             public EntityLivingBase field_151366_a;
+@@ -482,7 +505,8 @@
+     static final class ModifierLiving implements EnchantmentHelper.IModifier
+         {
+             public float field_77495_a;
+-            public EnumCreatureAttribute field_77494_b;
++            public @javax.annotation.Nullable Entity entity;
++            public @javax.annotation.Nullable EnumCreatureAttribute field_77494_b;
```

Comment:
`Nullable` should go on a new line before the `EnumCreatureAttribute` field, so the original line is not edited.
the new `entity` field should go after the vanilla fields

Reply:
Hmm, this field should be `EnumCreatureAttribute.UNDEFINED` by default instead of `null`.

Reply:
That's how mojang has it (null)... the way the class itself is used is very strange.

Reply:
Will change that.

Yeah, this has one of the strangest concepts I have seen in Minecraft so far